File: Advent of Code/2015/Day1/1.c
#include<stdio.h>

int main(void){
    int floor = 0, first_basement = 0, position = 1;
    while(!feof(stdin)){
        int character = getchar();
        if(character == 40)
            floor++;
        else if(character == 41)
            floor--;
        if(floor < 0 && first_basement == 0)
            first_basement = 1;
        else if(first_basement == 0)
            position++;
    }
    printf("Part one: %d\n", floor);
    printf("Part two: %d\n", position);
    return 0;
}

File: Advent of Code/2015/Day1/1.py
def main():
    data = input()
    print('Part one: {}'.format(data.count('(') - data.count(')')))
    count = 0
    for i in range(len(data)):
        if data[i] == '(':
            count += 1
        if data[i] == ')':
            count -= 1
        if count < 0:
            print('Part two: {}'. format(i+1))
            break

if __name__ == '__main__':
    main()